scipy/signal/_spectral_py.py

```diff
@@ -1987,14 +2025,29 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
 
     .. versionadded:: 0.16.0
     """
-    # Created sliding window view of array
+    xp = array_namespace(x)
+    x = xp.asarray(x)
+    # Created strided array of data segments
     if nperseg == 1 and noverlap == 0:
-        result = x[..., np.newaxis]
+        result = x[..., xp.newaxis]
     else:
         step = nperseg - noverlap
-        result = np.lib.stride_tricks.sliding_window_view(
-            x, window_shape=nperseg, axis=-1, writeable=True
-        )
+        # violating the array API standard gives us 4 orders of magnitude
+        # better performance via views/striding, so preserving some special
+        # casing
+        # see: https://github.com/data-apis/array-api/issues/641#issuecomment-1604884351
+        if is_cupy(xp):
+            result = xp.lib.stride_tricks.sliding_window_view(
+                x, window_shape=nperseg, axis=-1, writeable=True
```

Comment:
Trying to re-run the benchmarks from the SciPy conference array API proceedings paper (https://github.com/data-apis/scipy-2023-presentation) with CuPy `13.1.0` (which actually has `sliding_window_view`) I see: 

```
  File "/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/lib/stride_tricks.py", line 119, in sliding_window_view
    raise NotImplementedError("Writeable views are not supported.")
NotImplementedError: Writeable views are not supported.
```